### ISSUE TITLE
Native Session support

### ISSUE THREAD

**Issue by andrew-azarov** [bottlepy/bottle#591]
As beaker is no longer maintained there might be a need for native session support with files.

This can be made in one class which should not take a lot of space in the module.
I can supply a not so working one (had problems with concurrency when using gevent for example)...
This is only feature which should be in bottle as currently it requires a dance with the wolves to set up correctly and beaker is still buggy as hell.

**Comment by keredson** [comment]
I don't think bringing it into bottle is the right answer, but the example using beaker's session should definitely be removed or replaced.  We got hit hard with it's race conditions last year, and wound up rolling our own replacement: https://github.com/keredson/drsession

**Comment by defnull** [comment]
Yes, beaker is not perfect. But I do not known any alternative that really solves this issue other than not using sessions and store stuff in a real database instead. Even 'drsession' shows the same race condition problem if you modify the same key in a session from two concurrent requests. For example, you would not be able implement a request counter (for rate limiting) with drsession in any reliable way. You'd have to access redis directly for that (using the redis `HINCRBY` command) or risk loosing updates occasionally.

I think it would not hurt to warn users about these limitations, and make absolutely clear that sessions are no substitute for a real database. If it's not acceptable to loose updates, don't put it in a session.

But for simple use cases, beaker and co are still useful. A typical workflow (store user info in session on login, remove session on logout) works just fine. I think its an error to use sessions for more than that.

**Comment by keredson** [comment]
Didn't see this reply before I responded to other issue, sorry.  Actually pretty sure we implemented hincrby for +=.  I don't agree that it's an error to use sessions for more than the session id.  But fair enough.

**Comment by keredson** [comment]
@defnull you were right, we weren't using `HINCRBY`.  Fixed. https://github.com/keredson/drsession/issues/3
